tests/test_authorisation.py

```diff
@@ -1,21 +1,54 @@
-from app.main.authorisation import get_local_authority_place_names
+import pytest
+from flask import g
+from fsd_utils.authentication.models import User
+from werkzeug.exceptions import Unauthorized
 
+from app.main.authorisation import check_authorised, get_local_authority_and_place_names
 
-def test_get_local_authority_place_names():
-    user1_places = get_local_authority_place_names("user@bolton.gov.uk")
-    user2_places = get_local_authority_place_names("user@newcastle-staffs.gov.uk")
-    user3_places = get_local_authority_place_names("user@wigan.gov.uk")
-    user4_places = get_local_authority_place_names("user@wmadeup.gov.uk")
+
+def test_custom_las_and_place_names(flask_test_client):
+    """Check that custom domains/emails are being added to the mapping used by authorization."""
+    assert "contractor@contractor.com" in flask_test_client.application.config["EMAIL_TO_LA_AND_PLACE_NAMES"]
+
+
+def test_check_authorized_success(flask_test_client):
+    valid_user = User(full_name="Test", roles=[], highest_role_map={}, email="user@wigan.gov.uk")
+
+    with flask_test_client.application.app_context():
+        g.user = valid_user
+        local_authorities, places = check_authorised()
+    assert local_authorities
+    assert places
+
+
+def test_check_authorized_failure(flask_test_client):
+    invalid_user = User(full_name="Test", roles=[], highest_role_map={}, email="unknown_user@unknown.gov.uk")
+
+    with pytest.raises(Unauthorized):
+        with flask_test_client.application.app_context():
+            g.user = invalid_user
+            check_authorised()
+
+
+def test_get_local_authority_place_names(flask_test_client):
+    with flask_test_client.application.app_context():
+        user1_places = get_local_authority_and_place_names("user@bolton.gov.uk")
+        user2_places = get_local_authority_and_place_names("user@newcastle-staffs.gov.uk")
+        user3_places = get_local_authority_and_place_names("user@wigan.gov.uk")
+        user4_places = get_local_authority_and_place_names("user@wmadeup.gov.uk")
+        user5_places = get_local_authority_and_place_names("contractor@contractor.com")
 
     assert user1_places == (
-        "Farnworth",
-        "Bolton",
+        ("Bolton Metropolitan Borough Council",),
+        (
+            "Farnworth",
+            "Bolton",
+        ),
     )
     assert user2_places == (
-        "Newcastle-Under-Lyme Town Centre",
-        "Kidsgrove",
-        "Newcastle-under-Lyme",
-        "Newcastle-under-Lyme Town Centre",
+        ("Newcastle-under-Lyme District Council",),
+        ("Newcastle-Under-Lyme Town Centre", "Kidsgrove", "Newcastle-under-Lyme", "Newcastle-under-Lyme Town Centre"),
     )
-    assert user3_places == ("Wigan",)
-    assert user4_places is None
+    assert user3_places == (("Wigan Metropolitan Borough Council",), ("Wigan",))
+    assert user4_places == (None, (None,))
```

Comment:
Why is this data structure a slightly different shape to the sucessfull result?

Reply:
Good spot

Reply:
I've actually changed this to `(None, None)`. Each value can either be a tuple of strings or None. A tuple containing None suggests that it's returning a tuple of LAs where one takes the value of None, whereas just a None value is clearer at suggesting there are no LAs.